### ISSUE TITLE
[react-native] useInfiniteQuery initialPageParam is not set 

### ISSUE THREAD

**Issue by advancedtw** [TanStack/query#6027]
### Describe the bug

in react-native, despite setting an initialPageParams the **first query**  **pageParams** is always undefined. this bug doesn't happen with nextjs. I've tested with beta.0 , beta.28, beta.31
```tsx
useInfiniteQuery({
    queryKey: ["my-key"],
    queryFn: async ({pageParam}) => {
      console.log(pageParam, 'first query is always undefined ')
      const response = await fetch(`https://dummyjson.com/products?limit=12&skip=${pageParam}`)
      const data = (await response.json()) 
      return data.products
    },
    initialPageParam: 12, // <== doesn't matter what the value is. doesn't get passed to queryfn
    getPreviousPageParam: (firstPage, allPages, firstPageParam) => {
      console.log('get prev always undefined ', firstPageParam)
      return 0
    },
    getNextPageParam: (lastPage, allPages, lastPageParam) => {
      console.log('get next always undefined ', lastPageParam)
      return lastPage?.nextCursor
    },
    maxPages: 3,
  })
  ```

### Your minimal, reproducible example

https://snack.expo.dev/@advancedtw/use-query-react-native-

### Steps to reproduce

run reproduction and check the logs.
expo snack seems to have issue with installing dependencies.  so it's better to run the repro locally. 

### Expected behavior

I'm expecting to get a pageParam that matches initialPageParam

### How often does this bug happen?

None

### Platform

wsl2 , android, react native 0.72, expo 49

### Tanstack Query adapter

None

### TanStack Query version

beta

### TypeScript version

5.2.2

### Additional context

The same issue doesn't happen with nextjs so it could be a bundling issue

**Comment by TkDodo** [comment]
are you sure you have the correct version of RQ installed? Somewhen in `beta`, we renamed `defaultPageParam` to `initialPageParam`. If you have an older beta version, that would explain it. There shouldn't be any difference otherwise between React Native and nextJs.

> expo snack seems to have issue with installing dependencies. so it's better to run the repro locally.

not sure how I'd do that

**Comment by advancedtw** [comment]
yes I'm aware of the rename I've seen the pr. also I don't get a type error when I set initialPageParam. I've also checked my nodes_modules (yarn classic)  it's definitely using 5.0.0-beta.31.  

> There shouldn't be any difference otherwise between React Native and nextJs.

react native projects usually use metro bundler I'm guessing it's possible there is some slight difference in configuration or the way modules are getting picked up. or how the code gets transformed by babel to be fed into hermes.

regardless I have created an other clean project using bun instead of yarn and added the bare minimum dependencies and also removed ts support from the app.  **initialPageParam is set as expected** so I will close the issue but keep experimenting with different config. if I find something worth documenting I will let you know. 

many thanks
